x-pack/plugins/maps/public/components/map/feature_tooltip.js

```diff
@@ -11,52 +11,101 @@ import { i18n } from '@kbn/i18n';
 
 export class FeatureTooltip extends React.Component {
 
+  _renderFilterButton(tooltipProperty, hasFilters) {
+    if (!this.props.isFilterable || !hasFilters)  {
+      return null;
+    }
 
-  _renderProperties() {
-    return Object.keys(this.props.properties).map(propertyName => {
+
+    let icon;
+    if (tooltipProperty.isFilterable()) {
+      icon = (<EuiButtonIcon
+        iconType="plusInCircle"
+        title="Filter on property"
+        onClick={() => {
+          this.props.closeTooltip();
+          const filterAction = tooltipProperty.getFilterAction();
+          filterAction();
+        }}
+        aria-label="Filter on property"
+        className="mapFeatureTooltipFilterButton"
+      />);
+    } else {
+      icon = null;
+    }
+
+
+    return (
+      <EuiFlexItem className="mapFeatureTooltipFilterButton" grow={false}>
+        {icon}
+      </EuiFlexItem>
+    );
+  }
+
+  _renderProperties(hasFilters) {
```

Comment:
parameter `hasFilters` is called `hasFilterableProperties` by the caller. I like `hasFilterableProperties` because its more descriptive. Could you change `hasFilters` to `hasFilterableProperties` in this method and _renderFilterButton?

Reply:
by removing the empty flex-item for that case, it is no longer needed, so I removed it.